tasks/Logger.cpp

```diff
@@ -56,13 +61,38 @@ Logger::~Logger()
 
 bool Logger::startHook()
 {
-    if (_file.value().empty())
-        return false;
+    if(_file.value().empty())
+    {
+      log(Error) << "Could not create log file. Task property _file is empty." << endlog();
+      return false;
+    }
 
-    if(boost::filesystem::exists(_file.value()) && !_overwrite_existing_files.get())
+    if(_overwrite_existing_files.get() && _auto_rename_existing_files.get())
     {
-        log(Error) << "File " << _file.value() << " already exists." << endlog();
-        return false;
+      log(Error) << "Ambiguous property values. Both _overwrite_existing_files and _auto_rename_existing_files are set to true." << endlog();
+      return false;
+    }
+
+    if(boost::filesystem::exists(_file.value()) && !_overwrite_existing_files.get() && _auto_rename_existing_files.get())
+    {
+        log(Warning) << "File " << _file.value() << " already exists." << endlog();
+        // create timestamp
+        time_t now = time(0);
+        tm *t_ptr = localtime(&now);
+        char suffix[21];
+        strftime(suffix, sizeof(suffix), "%F_%H-%M-%S", t_ptr);
+        // append suffix to previous _file.value()
+        vector<string> strs;
+        boost::split(strs, _file.value(), boost::is_any_of("."));
+        strs.insert(strs.end()-1, std::string(suffix));
+        // safety check if timestamped file exists
+        std::string timestamped_str = boost::algorithm::join(strs, ".");
+        if(boost::filesystem::exists(timestamped_str)) {
+          log(Error) << "Timestamped file " << _file.value() << " already exists. Please retry." << endlog();
+          return false;
+        }
+        _file.set(timestamped_str);
+        log(Warning) << "Writing to " << _file.value() << " ." << endlog();
```

Comment:
Why would this be a warning ? Isn't this Info at best ?

Reply:
We can change to Info. My original thought was that changing the property from within the code may be confusing and the user should be warned about the automatic renaming. The automatic_renaming_* property makes the warning dispensable, but I would say it is worth an Info to notify about the final log file name.